js/modules/popupRegistration.js

```diff
@@ -0,0 +1,76 @@
+const body = document.body;
+const popupLinks = document.body.querySelectorAll(".popup-link");
+const registerModalOverlay = document.querySelector(".register__modal_overlay");
+const registerModalContent = registerModalOverlay.querySelector(".register__modal_content");
+const registerModalGratitude = registerModalOverlay.querySelector(".register__modal_gratitude");
+const closeBtn = registerModalOverlay.querySelector(".register__close_btn");
+const form = registerModalOverlay.querySelector(".register__form");
+const modalSubmitBtn = registerModalOverlay.querySelector(".frm__btn");
+const scrollBarWidth = window.innerWidth - document.documentElement.clientWidth;
+const REOPEN_DELAY = 800;
+const SHOW_TIME = 3000;
+const BTN_ESC = "Escape";
+let isOpen = true;
+
+for(let index = 0; index < popupLinks.length; index++){
+	const popupLink = popupLinks[index];
+
+	popupLink.addEventListener("click", function(){
+		popupOpen();
+		addButtonContent(popupLink);
+	})
+}
+
+function popupOpen(){
+	if(!isOpen) return;
+
+	registerModalOverlay.classList.add("open");
+	registerModalContent.classList.add("open");
+	body.classList.add("stopPageScroll");
+	body.style.paddingRight = scrollBarWidth + "px";
+	isOpen = false;
+
+	registerModalOverlay.addEventListener("click", function(e){
+		if(!e.target.closest(".register__modal_content")){
+			popupClose(registerModalContent);
+		}
+	})
+	closeBtn.addEventListener("click", function(){
+		popupClose(registerModalContent);
+	})
+	document.addEventListener("keydown", function(e){
+		if(e.code === BTN_ESC){
+			popupClose(registerModalContent);
+		}
+	})
+	setTimeout(() => isOpen = true, REOPEN_DELAY);
+
+	form.addEventListener("submit", function(e){
```

Comment:
Please, place this outside the function and use normal name for parameters, "event" instead of "e"

Reply:
Done.